### DISCUSSION TITLE
Using KafkaListener but facing issue to convert the content of the message

### DISCUSSION THREAD

**Discussion by alexandreab1202** [spring-projects/spring-kafka#2533] [General]
Noob with Kafka here! Please tell me if this question is on the wrong place and I can delete it.

I've been using the `kafka-protobuf-console-producer` to publish a message to a topic with a proto file registered and the command like looks like it worked!
<img width="1169" alt="image" src="https://user-images.githubusercontent.com/119886946/210638013-a9622bd3-19b2-4fbb-8736-701daa5acfe3.png">
My code works when I am using a simple java.lang.String but when I change to my specific Object (protocol buffer generated kotlin class) I get this issue.

But when my Kotlin with Spring Boot application using the `@KafkaListener()` try to consume the topic I get this message and I don't know what I am doing wrong:
`MessageConversionException: Cannot convert from [java.lang.String] to [com.MyProtoOrder$Items] for GenericMessage...`

Could someone please, share Kotlin or Java examples using this listener to convert the message sent by the producer to a specific object or tell me what I am doing wrong in my application? Probably my mistake is in this Kotlin code with my object and the proto file that I am using when I produce the protocol buffer message?
`fun consumeIngestion(myOrder: Order.Items) {
...
}`

**Comment by garyrussell** [comment]
I am not familiar with protobuf, but it looks like you are using a StringDeserializer - you either need a specific deserializer in your consumer configuration to decode the protobuff, or you need to receive it as a String or `byte[]` in your listener and convert to the protobuff object there (or use a custom `MessageConverter`).

You should ask your question to a wider audience, familiar with using prototbuf with Kafka, perhaps on Stack Overflow.

**Reply by alexandreab1202** [reply]
awesome! thank you

**Reply by alexandreab1202** [reply]
hey @garyrussell 
The issue that I shared with you is fixed. I needed add the Kotlin class like is [documented here](https://docs.spring.io/spring-kafka/reference/html/#spring-boot-consumer-app).

**Reply by artembilan** [reply]
Sorry, what class? The doc you show just official Spring for Apache Kafka reference manual and the class which jumps to the eyes is just plain main for Spring Boot. I fully lost how is this relevant to your problem with Protobuf…
